UpsertedProfileOrchestrator/handler.ts

```diff
@@ -105,14 +105,27 @@ export const getUpsertedProfileOrchestratorHandler = (params: {
       name
     } = upsertedProfileOrchestratorInput;
 
-    const profileOperation = oldProfile !== undefined ? "UPDATED" : "CREATED";
+    type ProfileOperation =
+      | {
+          readonly type: "UPDATED";
+          readonly oldProfile: RetrievedProfile;
+        }
+      | {
+          readonly type: "CREATED";
+        };
+
+    const profileOperation: ProfileOperation =
+      oldProfile !== undefined
+        ? { oldProfile, type: "UPDATED" }
+        : { type: "CREATED" };
 
     // Check if the profile email is changed
     const isProfileEmailChanged =
-      profileOperation === "UPDATED" && newProfile.email !== oldProfile.email;
+      profileOperation.type === "UPDATED" &&
+      newProfile.email !== profileOperation.oldProfile.email;
     // NOTE: if the following check is changed make sure to pass add the name field
     // to CreateProfile method which also calls this orchestrator
-    if (isProfileEmailChanged) {
+    if (isProfileEmailChanged && newProfile.email) {
```

Comment:
Although it's unlikely, if the email changes to undefined, could we raise an event?

Reply:
Do you mean just for monitoring?